### DISCUSSION TITLE
Renovate auth failure results in successful exit

### DISCUSSION THREAD

**Discussion by blitz** [renovatebot/renovate#11112] [Request Help]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [x] GitLab self-hosted

Renovate version: .15.17.1

**Describe the bug**

When Renovate fails to authenticate to Gitlab, it's exit code is 0, indicating success. This can make it hard to detect if the renovate setup is working without parsing logs.

The expected behavior is that authentication errors or other errors that abort execution result in a non-zero return value.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
% npm run renovate
> renovate-bot@0.0.0 renovate /home/julian/src/own/rust-renovate-example/renovate
> renovate

FATAL: Fatal error: No authentication found for platform https://gitlab.somewhere.de/api/v4/ (gitlab)
       "err": {
         "message": "No authentication found for platform https://gitlab.somewhere.de/api/v4/ (gitlab)",
         "stack": "Error: No authentication found for platform https://gitlab.somewhere.de/api/v4/ (gitlab)\n    at Object.parseConfigs (/home/julian/src/own/rust-renovate-example/renovate/node_modules/renovate/lib/config/index.js:98:11)\n    at async Object.start (/home/julian/src/own/rust-renovate-example/renovate/node_modules/renovate/lib/workers/global/index.js:22:18)\n    at async /home/julian/src/own/rust-renovate-example/renovate/node_modules/renovate/lib/renovate.js:7:3"
       }
% echo $?
0

```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

It's reproducible by running Renovate with this config.js:

```javascript
module.exports = {
  platform: 'gitlab',
  endpoint: 'https://gitlab.somewhere.de/api/v4/',
  token: process.env.RENOVATE_TOKEN,
};
```

Afterwards:

```
% npm run renovate
```

**Additional context**

...

**Comment by viceice** [comment]
Which version are you using? ` .15.17.1` isn't valid `renovate-bot@0.0.0` loocks wrong too.

Add `LOG_LEVEL=debug` to env

**Comment by viceice** [comment]
`No authentication found for platform` is no longer part of renovate, so i assume you are using a very ancient version. Please retry with a current v25 version of renovate.

**Reply by blitz** [reply]
True. I was following: https://simplabs.com/blog/2019/04/24/dependency-updates-for-gitlab/

Renovate did update itself to v25 and the problem is now gone. Sorry for the noise. :)

**Reply by viceice** [reply]
Thats outdated. Maybe @rarkins can add a blog post about our gitlab templates ? 🙃

**Reply by blitz** [reply]
@viceice Any pointers to best practices around a Gitlab setup would indeed be helpful!

**Reply by viceice** [reply]
https://gitlab.com/renovate-bot/renovate-runner
